### PR TITLE
Register PointMaze_Open_Diverse_GR

### LINKED ISSUE #194: [Bug Report] PointMaze_Open_DiverseGR not registered
If you are submitting a bug report, please fill in the following details and use the tag [bug].

**Describe the bug**
In `gymnasium_robotics.envs.maps` there is a map defined: 
```python
OPEN_DIVERSE_GR = [
    [1, 1, 1, 1, 1, 1, 1],
    [1, C, C, C, C, C, 1],
    [1, C, C, C, C, C, 1],
    [1, C, C, C, C, C, 1],
    [1, 1, 1, 1, 1, 1, 1],
]
```

However, the corresponding Gymnasium environment is not registered. 

Error message:
``` bash
gymnasium.error.NameNotFound: Environment `PointMaze_Open_Diverse_GR` doesn't exist. Did you mean: `PointMaze_Open_Diverse_G`?
```

**Code example**
```
import gymnasium as gym

env = gym.make('PointMaze_Open_Diverse_GR') # throws error

```

**System Info**
Gymnasium: `gymnasium_robotics==1.2.3` installed through `pip`
System: `Ubuntu 20.04`

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/gymnasium-robotics/issues) in the repo (**required**)

### PR DESCRIPTION
**Pull request by dtch1997**
# Description

Registers the `PointMaze_Open_Diverse_GR` environment. 

Fixes #194 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Kallinteris-Andreas** [comment]
Can you parameterize the registration of the maze environments, something like:
```py
for version in ["v3", "v4"]
    for type, entry_suffix in [("AntMaze", "ant_maze_{version}:AntMazeEnv"), ("PointMaze", "point_maze_{version}:PointMazeEnv")]:
        register(
            id=f"{type}_UMaze{suffix}-v3",
            entry_point=f"gymnasium_robotics.envs.maze.{entry_suffix}",
            kwargs=_merge(
                {
                    "maze_map": maps.U_MAZE,
                },
                kwargs,
            ),
            max_episode_steps=300,
        )
        ...
```

so that this is less likely to happen again

File: gymnasium_robotics/__init__.py [hunk 1]
```diff
@@ -1111,6 +1111,18 @@ def _merge(a, b):
...
+        register(
+            id=f"PointMaze_Open_Diverse_GR{suffix}-v3",
+            entry_point="gymnasium_robotics.envs.maze.point_maze:PointMazeEnv",
+            kwargs=_merge(
+                {
+                    "maze_map": maps.OPEN_DIVERSE_GR,
+                },
+                kwargs,
+            ),
+            max_episode_steps=300,
```
Comment (Kallinteris-Andreas) [inline gymnasium_robotics/__init__.py:1123]: 300 time steps is the same as `OPEN_DIVERSE_G`

**Review by Kallinteris-Andreas — APPROVED** [review]
@rodrigodelazcano do we want to add environments versions without a version bump?

**Comment by Kallinteris-Andreas** [comment]
Also, would like to point out that, `maps.EMPTY_MAZE` is not used for neither `PointMaze` nor `AntMaze`
https://github.com/Farama-Foundation/Gymnasium-Robotics/blob/f2754e19011a6ef329a40efc92dfa4bad81a22d3/gymnasium_robotics/envs/maze/maps.py#L19-L25

**Comment by Kallinteris-Andreas** [comment]
@dtch1997 please run the `pre-commit` checks

**Comment by dtch1997** [comment]
Hi @Kallinteris-Andreas thanks for getting back to me on the PR. 

I believe the failing pre-commit check is unrelated to my changes. Looking at the stack trace, the pyright failures appear to be due to this file: 
`/home/runner/work/Gymnasium-Robotics/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py`

I do not know how to fix these issues.

Full stack trace from [here](https://github.com/Farama-Foundation/Gymnasium-Robotics/actions/runs/7173616237/job/19709704219?pr=195):
```
typeshedPath /home/runner/work/Gymnasium-Robotics/Gymnasium-Robotics/typeshed is not a valid directory.
/home/runner/work/Gymnasium-Robotics/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py
  /home/runner/work/Gymnasium-Robotics/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:277:24 - error: "mujoco" is possibly unbound (reportUnboundVariable)
  /home/runner/work/Gymnasium-Robotics/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:278:23 - error: "mujoco_utils" is possibly unbound (reportUnboundVariable)
  /home/runner/work/Gymnasium-Robotics/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:308:9 - error: "mujoco" is possibly unbound (reportUnboundVariable)
  /home/runner/work/Gymnasium-Robotics/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:355:27 - error: "mujoco_py" is possibly unbound (reportUnboundVariable)
  /home/runner/work/Gymnasium-Robotics/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:356:23 - error: "mujoco_py_utils" is possibly unbound (reportUnboundVariable)
```

**Comment by Kallinteris-Andreas** [comment]
@dtch1997 does it pass `pre-commit` on your computer?


I am looking into the CI issues

**Comment by dtch1997** [comment]
@Kallinteris-Andreas No, I have the same error when running the linters locally. 

```
$ pre-commit run --all-files
black....................................................................Passed
codespell................................................................Passed
flake8...................................................................Passed
isort....................................................................Passed
pyupgrade................................................................Passed
pyright..................................................................Failed
- hook id: pyright
- exit code: 1

typeshedPath /home/daniel/Documents/github/Gymnasium-Robotics/typeshed is not a valid directory.
/home/daniel/Documents/github/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py
  /home/daniel/Documents/github/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:277:24 - error: "mujoco" is possibly unbound (reportUnboundVariable)
  /home/daniel/Documents/github/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:278:23 - error: "mujoco_utils" is possibly unbound (reportUnboundVariable)
  /home/daniel/Documents/github/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:308:9 - error: "mujoco" is possibly unbound (reportUnboundVariable)
  /home/daniel/Documents/github/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:355:27 - error: "mujoco_py" is possibly unbound (reportUnboundVariable)
  /home/daniel/Documents/github/Gymnasium-Robotics/gymnasium_robotics/envs/robot_env.py:356:23 - error: "mujoco_py_utils" is possibly unbound (reportUnboundVariable)
5 errors, 0 warnings, 0 informations

```

**Pull request merged**